.cicd/platforms/pinned/oracle_linux-8.3-pinned.dockerfile

```diff
@@ -0,0 +1,87 @@
+    cd build && \
+    cmake -G 'Unix Makefiles' -DLLVM_TARGETS_TO_BUILD=host -DLLVM_BUILD_TOOLS=false -DLLVM_ENABLE_RTTI=1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_TOOLCHAIN_FILE='/tmp/clang.cmake' -DCMAKE_EXE_LINKER_FLAGS=-pthread -DCMAKE_SHARED_LINKER_FLAGS=-pthread -DLLVM_ENABLE_PIC=NO -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_ENABLE_Z3_SOLVER=OFF .. && \
+    make -j$(nproc) && \
+    make install && \
+    cd / && \
+    rm -rf /llvm
+
+# build erlang from source (required for rabbitmq)
+RUN curl -LO https://github.com/erlang/otp/releases/download/OTP-23.3.4.7/otp_src_23.3.4.7.tar.gz && \
+    tar -xzvf otp_src_23.3.4.7.tar.gz && \
+    cd otp_src_23.3.4.7 && export ERL_TOP=`pwd` && \
+    ./configure && make -j$(nproc) && make install && \
+    cd .. && rm -rf otp_src_23.3.4.7 otp_src_23.3.4.7.tar.gz
+
+# manually install rabbitmq from binary 
+RUN dnf update && dnf install -y xz && \
+    curl -LO https://github.com/rabbitmq/rabbitmq-server/releases/download/v3.9.5/rabbitmq-server-generic-unix-3.9.5.tar.xz && \
+    tar -xf rabbitmq-server-generic-unix-3.9.5.tar.xz && \
+    mv rabbitmq_server-3.9.5 /usr/share && \
+    rm rabbitmq-server-generic-unix-3.9.5.tar.xz && \
+    ln -s /usr/share/rabbitmq_server-3.9.5/sbin/rabbitmq-server  /usr/sbin/rabbitmq-server && \
+    ln -s /usr/share/rabbitmq_server-3.9.5/sbin/rabbitmq-plugins  /usr/sbin/rabbitmq-plugins && \
+    ln -s /usr/share/rabbitmq_server-3.9.5/sbin/rabbitmq-env  /usr/sbin/rabbitmq-env     
+
+# build doxygen
+RUN curl -LO https://github.com/doxygen/doxygen/archive/refs/tags/Release_1_9_2.tar.gz && \
+    tar -xzvf Release_1_9_2.tar.gz && \
+    cd doxygen-Release_1_9_2 && \
+    mkdir build && cd build && cmake -G "Unix Makefiles" .. && \
+    make -j$(nproc) && make install && \
+    cd / && rm -rf doxygen-Release_1_9_2 Release_1_9_2.tar.gz
+
+# build boost
+RUN curl -LO https://boostorg.jfrog.io/artifactory/main/release/1.72.0/source/boost_1_72_0.tar.bz2 && \
+    tar -xjf boost_1_72_0.tar.bz2 && \
+    cd boost_1_72_0 && \
+    ./bootstrap.sh --with-toolset=clang --prefix=/usr/local && \
+    ./b2 toolset=clang cxxflags='-stdlib=libc++ -D__STRICT_ANSI__ -nostdinc++ -I/usr/local/include/c++/v1 -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fpie' linkflags='-stdlib=libc++ -pie' link=static threading=multi --with-iostreams --with-date_time --with-filesystem --with-system --with-program_options --with-chrono --with-test -q -j$(nproc) install && \
+    cd / && \
+    rm -rf boost_1_72_0.tar.bz2 /boost_1_72_0
+
+# requests module. used by tests
+RUN python3 -m pip install requests
+
+# install nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
+
+# install node 10
+RUN bash -c '. ~/.bashrc; nvm install --lts=dubnium' && \
+    ln -s "/root/.nvm/versions/node/$(ls -p /root/.nvm/versions/node | sort -Vr | head -1)bin/node" /usr/local/bin/node
+RUN yum install -y nodejs && \
+    yum clean all && rm -rf /var/cache/yum
```

Comment:
We are installing `node` twice here. I believe our tests are using whatever the package manager is giving us, not what is being specified by NVM. We should use one or the other. The NVM install inside docker is pretty hacky because it was never designed for non-interactive shells. It looks like nodeJS [officially supports installing a specific LTS on RHEL-family systems](https://nodejs.org/en/download/package-manager/#centos-fedora-and-red-hat-enterprise-linux). We can do it that way.